### PR TITLE
Ignore equivocating and duplicate messages from a sender

### LINKED ISSUE #153: Drop equivocations
Prior versions of the protocol spec required a node to retain all equivocating messages (same instance, round, step and peer but different value). The code reflects this. The explicit justification removes that need, so the code should instead ignore equivocations. Change it to ignore all but the first message from a peer for each round and step (this will also help one angle of #12).

With luck, this can be done just by changing the quorum state data structures to be only able to represent one value per participant, and checking if such value has been received before updating.

### PR DESCRIPTION
**Pull request by anorth**
Closes #153

### CODE REVIEW

File: gpbft/gpbft.go [hunk 1]
```diff
@@ -890,39 +891,63 @@ func newQuorumState(powerTable PowerTable) *quorumState {
...
+	}
+	q.receiveInner(sender, value, senderPower, signature)
+}
+
+// Receives each prefix of a chain as a distinct value from a sender.
+// Note that this method does not store signatures, so it is not possible later to
+// create an aggregate for these prefixes.
+// This is intended for use in the QUALITY phase.
+// Ignores any subsequent values from a sender from which a value has already been received.
+func (q *quorumState) ReceiveEachPrefix(sender ActorID, values ECChain) {
```
Comment (Stebalien) [inline gpbft/gpbft.go:909]: Did you mean to use this? Or is this for a future patch?

File: gpbft/gpbft.go [hunk 2]
```diff
@@ -1090,22 +1122,29 @@ type ConvergeTicket struct {
...
 
-	// !!! TODO: Drop duplicates (see https://github.com/filecoin-project/go-f3/issues/12)
-	c.values[key] = ConvergeValue{Chain: value, Justification: justification}
-	c.tickets[key] = append(c.tickets[key], ConvergeTicket{Sender: sender, Ticket: ticket})
-
+	// Keep only the first justification and ticket received for a value.
+	if _, found := c.values[key]; !found {
+		c.values[key] = ConvergeValue{Chain: value, Justification: justification}
+		c.tickets[key] = append(c.tickets[key], ConvergeTicket{Sender: sender, Ticket: ticket})
+	}
```
Comment (Stebalien) [inline gpbft/gpbft.go:1147]: I think I missed this part of the conversation. The code looks correct, but I don't have a strong enough understanding of the protocol to give it an OK from that perspective.

**Review by Stebalien — APPROVED** [review]
LGTM with a caveat.

**Review by ranchalp — APPROVED** [review]
LGTM!

File: gpbft/gpbft.go (hunk 2 above)
Comment (ranchalp) [reply gpbft/gpbft.go:1147]: This is ok!

File: gpbft/gpbft.go (hunk 1 above)
Comment (anorth) [reply gpbft/gpbft.go:909]: Thanks, good catch. A call to this replaces the equivalent code in the QUALITY_PHASE case in `receieveOne`.

**Pull request merged**